### PR TITLE
feat(appian-email): Add email alerting for compare-appian

### PR DESCRIPTION
**Pull request by charmcitygavin**
## Purpose

Adds an email alerting system for records that don't match between Appian and SEA Tool. Similar to the existing `compare-mmdl` service.

## Approach

Sends an email to specified users when there is no matching record for an Appian record in SEA Tool. Changes the email text to reflect that it's urgent if it has been longer than five days. Works the same way as `compare-mmdl`.

### CODE REVIEW

File: src/services/compare-appian/handlers/sendNoMatchAlert.ts [hunk 1]
```diff
@@ -21,80 +37,79 @@ exports.handler = async function (
...
 
-  const data = { ...event.Payload };
+  const data = { ...event.Payload } as Types.AppianSeatoolCompareData;
   const id: string = data.SPA_ID;
+  const secretExists = await Libs.doesSecretExist(region, secretId);
+  const secSinceAppianSubmitted = data.secSinceAppianSubmitted || 0;
 
-  const secretExists = await doesSecretExist(region, secretId);
+  // Was this submitted more than five days ago? If so, it's urgent:
+  const isUrgent = secSinceAppianSubmitted >= 432000; // Five days in secs
```
Comment (jdinh8124) [inline src/services/compare-appian/handlers/sendNoMatchAlert.ts:48]: Was this a new business requirement from Jonya?

File: src/services/compare-appian/serverless.yml [hunk 2]
```diff
@@ -157,14 +157,6 @@ functions:
...
       namespace: ${self:service}-${sls:stage}
-  sendDoesNotExistAlert:
-    handler: handlers/sendDoesNotExistAlert.handler
-    environment:
-      region: ${self:provider.region}
-      project: ${self:custom.project}
-      stage: ${sls:stage}
-      sesLogGroupName: ${param:sesLogGroupName}
-      errorsTopicArn: ${param:errorsTopicArn}
   sendNoMatchAlert:
```
Comment (jdinh8124) [inline src/services/compare-appian/serverless.yml:160]: You will also need to change line 262 from

`            Resource: !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:${self:service}-${sls:stage}-sendDoesNotExistAlert"
`

To

`            Resource: !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:${self:service}-${sls:stage}-sendNoMatchAlert"
`

**Review by jdinh8124 — COMMENTED** [review]
Looks good, a few comments. Were you also able to receive emails testing using the secrets?

File: src/services/compare-appian/handlers/sendNoMatchAlert.ts (hunk 1 above)
Comment (charmcitygavin) [reply src/services/compare-appian/handlers/sendNoMatchAlert.ts:48]: It appeared to be, judging from the email templates. But I'm not sure. That's a great question, I'll talk to Dany to get confirmation from her.

File: src/services/compare-appian/serverless.yml (hunk 2 above)
Comment (charmcitygavin) [reply src/services/compare-appian/serverless.yml:160]: Ah, great catch!

**Comment by charmcitygavin** [comment]
@jdinh8124 I haven't tested it yet. I'll make the above changes and then give it a shot. I may need some guidance to get it set up, unless we have some testing docs that could help

**Review by jdinh8124 — APPROVED** [review]
Looks good to me

File: src/services/compare-appian/handlers/utils/getEmailContent.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        <br>
+        <div style=' background-color: rgb(22, 82, 150); width: 580px; padding: 20px; margin: 20px; color: white;'> if you have any questions, please contact the help desk at <a href="mailto:SEATool_helpDesk@cms.hhs.org" style="color:#ffffff;">SEATool_helpDesk@cms.hhs.org</a> </div>
+        </center>
+      </body>
+    </html>
+  `;
+
+  const textData = `
+    This is a reminder that there's no matching record in SEA Tool for ${id}. Either a record wasn't created in SEA Tool, or the SPA ID in MMDL and SEA Tool don't match.
```
Comment (benjaminpaige) [inline src/services/compare-appian/handlers/utils/getEmailContent.ts:35]:
```suggestion
    This is a reminder that there's no matching record in SEA Tool for ${id}. Either a record wasn't created in SEA Tool, or the SPA ID in Appian and SEA Tool don't match.
```

Comment (benjaminpaige) [reply src/services/compare-appian/handlers/utils/getEmailContent.ts:35]: super small thing

**Review by benjaminpaige — APPROVED** [review]
small comment, but looks great.

**Comment by charmcitygavin** [comment]
We are still waiting on confirmation from Jonya about whether or not the current email template is correct, and whether or not we need a special "urgent" variation of the email, similar to MMDL.

**Pull request merged**